### PR TITLE
[Do-not-merge] Update pnp convention-based samples

### PR DESCRIPTION
**Pull request by abhipsaMisra**
Will be built and merged once the updated nugets have been released.

### CODE REVIEW

**Comment by abhipsaMisra** [comment]
Will be merged once the updated nugets have been released.

File: iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/Program.cs [hunk 1]
```diff
@@ -85,21 +85,6 @@ public static async Task Main(string[] args)
             await deviceClient.CloseAsync();
         }
 
-        private static ILogger InitializeConsoleDebugLogger()
```
Comment (drwill-ms) [inline iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/Program.cs:88]: Why removed?

**Review by drwill-ms — APPROVED** [review]

File: iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/SystemTextJsonPayloadSerializer.cs [hunk 2]
```diff
@@ -37,50 +36,15 @@ public override T DeserializeToType<T>(string stringToDeserialize)
...
         /// <inheritdoc/>
-        public override T ConvertFromObject<T>(object objectToConvert)
+        public override T ConvertFromJsonObject<T>(object jsonObjectToConvert)
         {
-            return DeserializeToType<T>(SerializeToString(objectToConvert));
+            return DeserializeToType<T>(SerializeToString(jsonObjectToConvert));
         }
 
         /// <inheritdoc/>
-        public override bool TryGetNestedObjectValue<T>(object nestedObject, string propertyName, out T outValue)
```
Comment (abhipsaMisra) [inline iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/SystemTextJsonPayloadSerializer.cs:46]: removed from the interface, so doesn't need to be implemented

**Pull request closed without merging**